src/pages/tutorial/setup-environment.mdx

```diff
@@ -0,0 +1,93 @@
+---
+tags: ["tutorial"]
+---
+
+import { Callout } from "nextra/components";
+
+# Environment setup
+
+## Rust installation
+
+To work with CosmWasm smart contract, you will need rust installed on your machine. If you don't
+have one, you can find installation instructions on
+[the Rust website](https://www.rust-lang.org/tools/install).
+
+I assume you are working with a stable Rust channel in this book.
+
+Additionally, you will need the Wasm rust compiler backend installed to build Wasm binaries. To
+install it, run:
+
+```shell copy filename="TERMINAL"
+rustup target add wasm32-unknown-unknown
+```
+
+## The cosmwasm-check utility
+
+An additional helpful tool for building smart contracts is the
+[`cosmwasm-check`](https://github.com/CosmWasm/cosmwasm/tree/main/packages/check) utility. It allows
+you to check if the wasm binary is a proper smart contract ready to upload into the blockchain. You
+can install it using cargo:
+
+```shell copy filename="TERMINAL"
+cargo install cosmwasm-check
+```
+
+If the installation succeeds, you should be able to execute the utility from your command line.
+
+```shell copy filename="TERMINAL"
+cosmwasm-check --version
+```
+
+The output should look like this:
+
+```shell
+Contract checking 1.2.3
+```
+
+## Verifying the installation
+
+To guarantee you are ready to build your smart contracts, you need to make sure you can build
+examples. Checkout the [`cw-plus`](https://github.com/CosmWasm/cw-plus) repository and run the
+testing command in its folder:
+
+```shell copy filename="TERMINAL"
+git clone git@github.com:CosmWasm/cw-plus.git && cd ./cw-plus && cargo test
+```
+
+You should see that everything in the repository gets compiled and all tests pass.
+
+THe [`cw-plus`](https://github.com/CosmWasm/cw-plus) is a great place to find example contracts -
```

Comment:
```suggestion
[`cw-plus`](https://github.com/CosmWasm/cw-plus) is a great place to find example contracts -
```
Proper name

Reply:
Other than capital `H`, it's intentional. I don't want to start the sentence with the lowercase.